internal/handlers/user.go

```diff
@@ -22,14 +22,24 @@ const AuthUserName string = "_nexodus.UserName"
 
 var noUUID = uuid.UUID{}
 
+// CacheExp Zero expiration means the key has no expiration time.
+const CacheExp time.Duration = 0
+
 func (api *API) CreateUserIfNotExists() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		id := c.GetString(gin.AuthUserKey)
 		username := c.GetString(AuthUserName)
-		_, err := api.createUserIfNotExists(c.Request.Context(), id, username)
+		cachedUsername, err := api.redis.Get(c.Request.Context(), id).Result()
```

Comment:
I would recommend adding a prefix to the cache key like "user:" or "create_user:" so that we can more confidently use the same Redis database as a cache for other things in the future.

Reply:
added